File: Latihan1/java/program/Main.java
/*  
	Saya Ibnu Adeng Kurnia NIM 2101769 mengerjakan latihan ke-1 
	dalam mata kuliah desain dan pemrograman berorientasi objek C2 2023
	untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. 
	Aamiin.
*/

/*  Design ini disusun dan/atau digunakan hanya untuk lingkungan sendiri.
	Tidak untuk menjadi konsumsi/kepentingan umum.
	Hanya untuk melengkapi tugas DPBO 2023.
*/

import java.util.Scanner;

class Main {
    public static void main(String[] args) {
         // ------------------------ // pendeklarasian // ------------------------ //
        Scanner sc = new Scanner(System.in);
        String namaFakultas = "";               // untuk nama fakultas //
        String prodi = "";                      // untuk program studi //
        int tahunMasuk = 0;                     // tahun masuk //
        DataPodik mhs = new DataPodik();
        int n = 0;
        int nipDosenpa = 0;                     // nip dosen PA //
        String namaDosenpa = "";                // nama dosen PA //
        
        // -------------------------------------------- HEADER -------------------------------------------------------------------------//
        System.out.println("\n                      KEMENTERIAN PENDIDIKAN KEBUDAYAAN RISET DAN TEKNOLOGI");
        System.out.println("                            PUSAT DATA DAN TEKNOLOGI INFORMASI (PUSDATIN)");
        System.out.println("                              PANGKALAN DATA PENDIDIKAN TINGGI (PDDIKTI)");
        System.out.println("                                DIREKTORAT JENDERAL PENDIDIKAN TINGI");
        System.out.println("                 Sekretariat Direktorat Jenderal Pendidikan Tinggi, Kementerian Pendidikan");
        System.out.println("                    Kebudayaan Riset dan Teknologi , Gedung D, Jalan Jenderal Sudirman,");
        System.out.println("                              Pintu 1 Senayan, Jakarta Pusat 10270");
        System.out.println("     Telp. 021-57946104, 021 741 8808. Email : pddikti@kemdikbud.go.id, pusdatin@kemdikbud.go.id");
        System.out.println(" =========================================================================================================");
        
        System.out.println("              Selamat Datang Bapak/Ibu diportal Pangkalan Data Pendidikan Tinggi (PDDikti)");
        System.out.println("                 PUSDATIN KEMENDIKBUDRISTEKDIKTI Direktorat Jenderal Pendidikan Tinggi");

        System.out.println(" ---------------------------------------------------------------------------------------------------------");
        System.out.println("                                      DATA POKOK PENDIDIKAN");
        System.out.println("                                    UNIVERSITAS IKATAN BATIN");
        System.out.println("                                 TAHUN PEMBAHARUAN 2022/2023-Genap");
        System.out.println(" ---------------------------------------------------------------------------------------------------------\n");
        
        
        // ==================== Menu Inputan/CREATE USER ======================== //
        System.out.println("\n MENU INPUTAN");
        System.out.println(" ISI Nama Fakultas  ");
        System.out.print("  > Nama Fakultas    : ");            // Isi nama fakultasnya //
        try {
            namaFakultas = sc.next();
        } catch (Exception e) {
        }
        System.out.print("  > Program Studi    : ");            // Isi nama program studinya //
        try {
            prodi = sc.next();
        } catch (Exception e) {
        }
        System.out.print("  > Tahun Masuk      : ");            // Isi tahun masuknya //
        try {
            tahunMasuk = sc.nextInt();
        } catch (Exception e) {
        }
        System.out.print("  > Jumlah Mahasiswa : ");            // Isi jumlah mahasiswa nya //
        try {
            n = sc.nextInt();
        } catch (Exception e) {
        }
        String[] mahasiswa = new String[n];
        System.out.println("  > Nama Mahasiswa : ");            // isi nama mahasiswanya //
        for (int i = 0; i < mahasiswa.length; i++) {
            mahasiswa[i] = "";
            System.out.print("    - ");
            try {
                mahasiswa[i] = sc.next();
            } catch (Exception e) {
            }
        }
        System.out.println("  > Dosen Pembimbing Akademik ");       // isi nama dosen pembimbing akademik //
        System.out.print("    Nama Dosen : ");
        try {
            namaDosenpa = sc.next();
        } catch (Exception e) {
        }
        System.out.print("    NIP        : ");                      // isi NIP dosen pembimbing akademik //
        try {
            nipDosenpa = sc.nextInt();
        } catch (Exception e) {
        }

        // setter dan/atau getter //
        mhs.setNamaFakultas(namaFakultas);
        mhs.setProdi(prodi);
        mhs.setMahasiswa(mahasiswa);
        mhs.setTahunMasuk(tahunMasuk);
        mhs.getDosenpa().setNamaDosenpa(namaDosenpa);
        mhs.getDosenpa().setnipDosenpa(nipDosenpa);

        // ===================== Menampilkan/read hasil dari yang diinputkan user ======================== //
        System.out.println("");
        System.out.println("\n Data Berhasil Ditambahkan!.");
        System.out.println("============================================");
        System.out.println("          DATA INFORMASI PENDIDIKAN\n   ");
        System.out.println("Nama Fakultas : " + mhs.getNamaFakultas());                 // menampilkan nama fakultas //
        System.out.println("Program Studi : " + mhs.getProdi());                        // menampilkan prodi //
        System.out.println("Tahun Masuk   : " + mhs.getTahunMasuk());                   // menampilkan tahun masuknya //
        System.out.println("\nDaftar Nama Mahasiswa ;");                                // menampilkan data mahasiswanya //
        for (int j = 0; j < mahasiswa.length; j++) {
            System.out.println((j + 1) + "." + mhs.getMahasiswa()[j]);
        }
        System.out.println("\nInformasi Dosen Pembimbing Akademik ;");                  // menampilkan data dosen PA //
        System.out.println("- Nama Dosen : " + mhs.getDosenpa().getNamaDosenpa());      // nama dosen PA //
        System.out.println("- NIP        : " + mhs.getDosenpa().getnipDosenpa());       // NIP dosen PA //
        System.out.println("============================================\n");

        // ----------------------------------------------- FOOTER --------------------------------------- //
        System.out.println("\n      Copyright 2023. IBNU ADENG KURNIA");
        System.out.println("\n             All Rights Reserved");
    }
}

File: Latihan1/py/program/main.py
''' 
	Saya Ibnu Adeng Kurnia NIM 2101769 mengerjakan latihan ke-1 
	dalam mata kuliah desain dan pemrograman berorientasi objek C2 2023
	untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. 
	Aamiin.
'''

'''  
    Design ini disusun dan/atau digunakan hanya untuk lingkungan sendiri.
	Tidak untuk menjadi konsumsi/kepentingan umum.
	Hanya untuk melengkapi tugas DPBO 2023.
'''

from DataPodik import *

## pendeklarasian ##
faks = []
namaFakultas = ""
prodi = ""
tahunMasuk = ""
n = 0

#### ================================================= HEADER ========================================= #######
print("\n                    KEMENTERIAN PENDIDIKAN KEBUDAYAAN RISET DAN TEKNOLOGI")
print("                          PUSAT DATA DAN TEKNOLOGI INFORMASI (PUSDATIN)")
print("                           PANGKALAN DATA PENDIDIKAN TINGGI (PDDIKTI)")
print("                              DIREKTORAT JENDERAL PENDIDIKAN TINGI")
print("             Sekretariat Direktorat Jenderal Pendidikan Tinggi, Kementerian Pendidikan")
print("                Kebudayaan Riset dan Teknologi , Gedung D, Jalan Jenderal Sudirman,")
print("                               Pintu 1 Senayan, Jakarta Pusat 10270")
print("     Telp. 021-57946104, 021 741 8808. Email : pddikti@kemdikbud.go.id, pusdatin@kemdikbud.go.id")
print(" =========================================================================================================\n")

print("               Selamat Datang Bapak/Ibu diportal Pangkalan Data Pendidikan Tinggi (PDDikti)")
print("                  PUSDATIN KEMENDIKBUDRISTEKDIKTI Direktorat Jenderal Pendidikan Tinggi")

print(" ---------------------------------------------------------------------------------------------------------")
print("                                        DATA POKOK PENDIDIKAN")
print("                                       UNIVERSITAS IKATAN BATIN")
print("                                   TAHUN PEMBAHARUAN 2022/2023-Genap")
print(" ---------------------------------------------------------------------------------------------------------\n")


#Menu Inputan untuk user ####
n = int(input(" Silakan Masukkan Jumlah Fakultas yang akan ditambahkan : "))            ## meminta masukan ke user untuk jml fakultas yg akan dicreate 
print(" + + + + + + + + + + + + + + + + + + + + +")
for i in range(n):
    fak = DataPodik()
    mahasiswa = []
    namaFakultas = input("   > Nama Fakultas " + str(i + 1) + "  : ")                   ## meminta masukkan ke user untuk nama fakultas 
    prodi = input("   > Program Studi    : ")                                           ## meminta masukkan ke user untuk  prodi
    tahunMasuk = input("   > Tahun Masuk      : ")                                      ## meminta masukkan ke user untuk  tahun masuk 
    m = 0
    m = int(input("   > Jumlah Mahasiswa : "))                                          ## meminta masukkan ke user untuk  jml mahasiswa
    print("   > Daftar Mahasiswa")                                                      ## menampilkan data mahasiswa
    for j in range(m):
        strP = input("     - ")
        mahasiswa.append(strP)
    
    tahunBerdiri = ""
    pimpinan = ""
    print("   > Detail Program Studi")                                                  ## menampilkan detail prodi 
    tahunBerdiri = input("     - Tahun Berdiri          : ")                            ## meminta masukkan ke user untuk tahun berdiri prodinya
    pimpinan = input("     - Pimpinan Program Studi : ")                                ## meminta masukkan ke user untuk pimpinan prodinya

    ########## SETTER DAN GETTTER ################
    ## set untuk (nama fakultas, prodi, tahun amsuk, mahasiswa)
    ## get untuk (tahun berdiri dan pimpinan prodi)
    fak.setNamaFakultas(namaFakultas)
    fak.setProdi(prodi)
    fak.setTahunMasuk(tahunMasuk)
    fak.setMahasiswa(mahasiswa)
    fak.getDataProdi().setTahunBerdiri(tahunBerdiri)
    fak.getDataProdi().setPimpinan(pimpinan)

    faks.append(fak)
    print(" + + + + + + + + + + + + + + + + + + + + +\n")

################################### menampilkan daftar fakultas yg user buat ###################################
print("\n## ## ## ## ## ## ## ## ## ## ## ## ## ##")
print("Daftar Fakultas Yang Anda Isi;")
for i in range(n):
    print(str(i + 1), ".", faks[i].getNamaFakultas())
print("## ## ## ## ## ## ## ## ## ## ## ## ## ##\n")


print("")
idFak = 0
##################################### untuk melihat informasi fakutas beserta lainnya dari yg user inputakan ################################3
print("Jika Anda Ingin Melihat detail informasi Fakultas beserta lainnya, ikuti langkah berikut: ")
idFak = int(input("Tampilkan Detail Informasi di Fakultas ke- : "))
print(" + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + ")
print("   Silakan, berikut data yang Anda minta ;\n")
print("    > Nama Fakultas : " + faks[idFak-1].getNamaFakultas())                                      # nama fakultas 
print("    > Program Studi : " + faks[idFak-1].getProdi())                                             ## program studi
print("    > Tahun Masuk   : " + faks[idFak-1].getTahunMasuk())                                         ## tahun masuk
print("    > Daftar Mahasiswa ;")                                                                         ### daftar mahasiswa
for p in faks[idFak-1].getMahasiswa():
    print("      - " + p)
print("    > Detail Program Studi ;")                                                                      ## detail prodi
print("      - Tahun Berdiri           : " + faks[idFak-1].getDataProdi().getTahunBerdiri())                ## tahun berdiri
print("      - Pimpinan Program Studi  : " + faks[idFak-1].getDataProdi().getPimpinan())                    ## pimpinan
print(" + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + \n\n")

############################ ======================== FOOTER ========================= ########################################################S
print("                                  Copyrigt © 2023. IBNU ADENG KURNIA")
print("                                         All Rights Reserved")